packages/wasm-utils/src/note/secrets.rs

```diff
@@ -0,0 +1,49 @@
+use ark_std::rand::rngs::OsRng;
+use arkworks_circuits::setup::mixer::{setup_leaf, setup_leaf_with_privates_raw};
+use arkworks_gadgets::prelude::ark_bls12_381::Fq as BlsFr;
+use arkworks_gadgets::prelude::ark_bn254::Fq as Bn254Fr;
+use arkworks_gadgets::prelude::*;
+use arkworks_utils::utils::common::Curve as ArkworksCurve;
+
+use crate::types::{Curve, OpStatusCode};
+use crate::utils::get_hash_params_x5;
+
+pub fn generate_secrets(exponentiation: usize, curve: Curve, rng: &mut OsRng) -> Result<Vec<u8>, OpStatusCode> {
+	let secrets: Vec<u8> = match (curve, exponentiation) {
```

Comment:
After the chat yesterday, we should bring back the `width` here.
The `width` in the note is regarding the leaf width, so it should be `5` here

Reply:
The leaf is always of width 5, for artworks, right? if not please say more about what makes width useful here, as generating secrets will just create the leaf

Reply:
The leaf could be generated using widths other than 5 (for example with the width of 3)